### PR TITLE
refactor(addon-charts): support noUncheckedIndexedAccess

### PR DESCRIPTION
**Pull request by splincode**

### CODE REVIEW

File: projects/addon-charts/components/line-chart/line-chart-hint.directive.ts [hunk 1]
```diff
@@ -64,20 +64,23 @@ export class TuiLineChartHint implements AfterViewInit {
...
     public getContext(index: number, _chart: TuiLineChart): readonly TuiPoint[] {
-        return this.computeContext(...this.charts.map(({value}) => value))[index];
+        return this.computeContext(...this.charts.map(({value}) => value))[index] || [];
     }
 
     // _chart is required by TuiLineDaysChartComponent that impersonates this directive
     public raise(index: number, _chart: TuiLineChart): void {
-        const current = this.charts.map((chart) => chart.value[index]);
+        const current: ReadonlyArray<readonly [number, number]> = this.charts.map(
+            (chart) => chart.value[index] ?? [0, 0],
```
Comment (waterplea) [inline projects/addon-charts/components/line-chart/line-chart-hint.directive.ts:73]: Would `as const` work on [0, 0]?

File: projects/addon-charts/utils/draw.ts [hunk 2]
```diff
@@ -10,7 +10,11 @@ export function tuiDraw(
     index: number,
     smoothing: number,
 ): string {
-    return smoothing
-        ? tuiDrawCurve(array, index, smoothing / COEFFICIENT)
-        : tuiDrawLine([array[index][0], array[index][1]]);
```
Comment (waterplea) [inline projects/addon-charts/utils/draw.ts:15]: Would `...(array[index] ?? [0, 0])` work?

File: projects/addon-charts/utils/control-point.ts [hunk 3]
```diff
@@ -4,18 +4,18 @@ import {tuiLineAngle} from './line-angle';
...
-    current: TuiPoint,
+    current?: TuiPoint,
     previous?: TuiPoint,
     next?: TuiPoint,
     reverse = false,
     smoothing = 0.2,
 ): TuiPoint {
-    const p = previous || current;
-    const n = next || current;
+    const p = previous || current || [0, 0];
```
Comment (waterplea) [inline projects/addon-charts/utils/control-point.ts:13]: Maybe we can add first line: `const fallback = current || [0, 0] as const` and use it down the line for readability?

File: projects/addon-charts/components/line-days-chart/line-days-chart-hint.directive.ts [hunk 4]
```diff
@@ -62,15 +62,17 @@ export class TuiLineDaysChartHint implements AfterContentInit {
         return this.getMap(...this.charts.map(({value}) => value)).get(String(day)) || [];
     }
 
-    public raise(day: TuiDay): void {
+    public raise(day?: TuiDay): void {
```
Comment (waterplea) [inline projects/addon-charts/components/line-days-chart/line-days-chart-hint.directive.ts:65]: Let's not make it optional and instead add early return where it is called. Same with onHover there

File: projects/addon-charts/components/line-days-chart/line-days-chart.component.ts [hunk 5]
```diff
@@ -217,12 +230,14 @@ export class TuiLineDaysChart implements AfterViewInit {
...
-                          array[index + 1].find((day) => !Number.isNaN(day[1])) || DUMMY,
+                          array[index + 1]?.find((day) => !Number.isNaN(day[1])) || DUMMY,
                       ],
             );
     }
 
-    private getDay(index: number): TuiDay {
-        return this.value[index - this.value[0][0].day + 1][0];
+    private getDay(index: number): TuiDay | undefined {
+        const start = this.value[0]?.[0];
```
Comment (waterplea) [inline projects/addon-charts/components/line-days-chart/line-days-chart.component.ts:239]: Let's add `?.day || 0` here so the next line is simpler

**Review by waterplea — APPROVED** [review]

File: projects/addon-charts/components/line-days-chart/line-days-chart.component.ts (hunk 5 above)
Comment (splincode) [reply projects/addon-charts/components/line-days-chart/line-days-chart.component.ts:239]: image

File: projects/addon-charts/components/line-chart/line-chart-hint.directive.ts (hunk 1 above)
Comment (splincode) [reply projects/addon-charts/components/line-chart/line-chart-hint.directive.ts]: fixed

File: projects/addon-charts/utils/draw.ts (hunk 2 above)
Comment (splincode) [reply projects/addon-charts/utils/draw.ts:15]: fixed

File: projects/addon-charts/utils/control-point.ts (hunk 3 above)
Comment (splincode) [reply projects/addon-charts/utils/control-point.ts]: fixed

File: projects/addon-charts/components/line-days-chart/line-days-chart-hint.directive.ts (hunk 4 above)
Comment (splincode) [reply projects/addon-charts/components/line-days-chart/line-days-chart-hint.directive.ts]: fixed

**Review by vladimirpotekhin — APPROVED** [review]

**Pull request merged**